server/src/main/java/org/elasticsearch/search/aggregations/metrics/max/MaxAggregator.java

```diff
@@ -69,7 +74,14 @@ public LeafBucketCollector getLeafCollector(LeafReaderContext ctx,
             final LeafBucketCollector sub) throws IOException {
         if (valuesSource == null) {
             return LeafBucketCollector.NO_OP_COLLECTOR;
```

Comment:
should we early terminate in this case too?

Reply:
if we can (no parent agg), yes